pkg/resources/resource.go

```diff
@@ -27,4 +29,20 @@ type FlavorResource struct {
 	Resource corev1.ResourceName
 }
 
+func (fr FlavorResource) MarshalJSON() ([]byte, error) {
```

Comment:
I think implementing String should be enough, and much simpler. Then, you can convert sets to slices with built-in: `sets.List(set)`